### PR TITLE
Allow domain services to create arbitrary private topics.

### PR DESCRIPTION
**Pull request by big-andy-coates**
fixes: https://github.com/specmesh/specmesh-build/issues/61

### Reviewer checklist
- [ ] Read the [contributing guide](https://github.com/mockito/mockito/blob/main/.github/CONTRIBUTING.md)
- [ ] PR should be motivated, i.e. what does it fix, why, and if relevant, how
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")
- [ ] Ensure any appropriate documentation has been added or amended

### CODE REVIEW

File: kafka/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+## Authorisation
+
+## Topic authorisation
+
+The provisioner will set ACLs to enforce the following topic authorisation:
+
+### Public topics:
+
+Only the domain itself can `WRITE` to public topics; Any domain can `READ` from them.
```
Comment (bluemonk3y) [inline kafka/README.md:13]: public topics are denoted using a relative path from the domain and prefixed with '_public'. The resulting topics will be domain.structure._public.xxx.yyy

File: kafka/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+### Protected topics:
+
+Only the domain itself can `WRITE` to protected topics; Any domain specifically tagged in the spec can `READ` from them.
```
Comment (bluemonk3y) [inline kafka/README.md:17]: similarly - some kind of example? wdyt?

**Review by bluemonk3y — APPROVED** [review]
minor comment on providing examples of how to specify _public, _private, and _protected topics and how they are 'relative' to the domain-id... lgtm

File: kafka/README.md (hunk 2 above)
Comment (big-andy-coates) [reply kafka/README.md:17]: Agreed, but outside the scope of this PR.  I'm just wanting to document what I did in this and the last PR around ACLs.

File: kafka/README.md (hunk 1 above)
Comment (big-andy-coates) [reply kafka/README.md:13]: I've added something, but again, this is outside the scope of what I was doing in this PR. 

We definitely need to write some docs on what we already have... https://github.com/specmesh/specmesh-build/issues/64

**Pull request merged**